### PR TITLE
Supermirror, flipper matrices, and workflow with configurable polarizer/analyzer

### LINKED ISSUE #59: Extend correction to include spin-flipper matrices
Follow up to #55.

### PR DESCRIPTION
**Pull request by SimonHeybrock**
This adds:

- Flipper matrices (seems to be working, according to tests).
- Supermirror skeleton. The actual efficiency function is not implemented.
- `PolarizationAnalysisWorkflow`, which composes workflows for the polarization and analyzer (each can be He3 or supermirror) into a single workflow.

Fixes #59.

### CODE REVIEW

**Comment by jokasimr** [comment]
It's hard to review this without a description of the procedure. I did not find anything out flippers in [the polarization analysis methodology page](https://github.com/scipp/esspolarization/blob/main/docs/user-guide/sans-polarization-analysis-methodology.ipynb). What did you use to implement this @SimonHeybrock?

**Comment by SimonHeybrock** [comment]
> It's hard to review this without a description of the procedure. I did not find anything out flippers in [the polarization analysis methodology page](https://github.com/scipp/esspolarization/blob/main/docs/user-guide/sans-polarization-analysis-methodology.ipynb). What source did you use to implement this @SimonHeybrock?

@astellhorn was adding information in #56, but I think a file was missing and the PR needs updating. Otherwise, have a look at the attachments in https://github.com/scipp/esspolarization/issues/2#issuecomment-1833647206 (probably not most recent version).

**Comment by astellhorn** [comment]
> > It's hard to review this without a description of the procedure. I did not find anything out flippers in [the polarization analysis methodology page](https://github.com/scipp/esspolarization/blob/main/docs/user-guide/sans-polarization-analysis-methodology.ipynb). What source did you use to implement this @SimonHeybrock?
> 
> @astellhorn was adding information in #56, but I think a file was missing and the PR needs updating. Otherwise, have a look at the attachments in [#2 (comment)](https://github.com/scipp/esspolarization/issues/2#issuecomment-1833647206) (probably not most recent version).

Yes, I forgot to add the new methodology file into the branch, and will update it (best today, but I am on a conference since sunday up to tomorrow, but will have some time today lunch break).

**Review by jokasimr — APPROVED** [review]
Seems fine to me! I just have one question about the flipper implementation.

File: src/ess/polarization/correction.py [hunk 1]
```diff
@@ -19,6 +23,53 @@
...
+    def from_left(
+        self, up: sc.Variable, down: sc.Variable
+    ) -> tuple[sc.Variable, sc.Variable]:
+        """Apply inverse flipper matrix from the left (for analyzer)"""
+        if self.swap:
+            up, down = down, up
+        f = 1 / self.efficiency.value
+        if f == 1:
+            return up, down
+        return up, (1 - f) * up + f * down
```
[the remark this answers is not in the document]
Comment (SimonHeybrock) [reply src/ess/polarization/correction.py:42]: Actually we dot have a vector `[u, d]` but a matrix `[[u, d], [d, u]]`. The function returns one of the resulting columns. Does it work out then?

[the remark this answers is not in the document]
Comment (jokasimr) [reply src/ess/polarization/correction.py:42]: Yes then it works out :+1:

**Pull request merged**